File: code/dobot_quad_sdk-main/high_level/python/examples/kill_robot.py
#!/usr/bin/env python3
"""Emergency shutdown: switch to PASSIVE then kill controller processes."""
import sys
from dobot_quad import RobotClient


def main():
    confirm = input("Kill robot controller? (y/N): ")
    if confirm.lower() != "y":
        print("Aborted.")
        return 0

    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    try:
        robot.execute("kill_robot")
    except Exception:
        print("Kill command sent (server shut down).")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: code/dobot_quad_sdk-main/high_level/cpp/kill_robot.cpp
// Kill Robot: Emergency shutdown (PASSIVE -> kill controller processes).
// Usage: ./kill_robot [server_address]

#include "robot_client.h"

int main(int argc, char** argv)
{
    std::string confirm;
    std::cout << "Kill robot controller? (y/N): ";
    std::cin >> confirm;
    if (confirm != "y" && confirm != "Y") {
        std::cout << "Aborted." << std::endl;
        return 0;
    }

    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    auto req = robot::make_request("kill");
    req.mutable_sequence()->add_motions()->set_motion_id("kill_robot");
    client.execute(req);
    return 0;
}
